### ISSUE TITLE
AdaBoost（boosting的一种）

### ISSUE THREAD

**Issue by necogy** [data-com/discussions#15]
ensemble method在当前机器学习领域备受欢迎的理由：灵活性（bias相对较低）和对样本变化的相对不敏感（variance相对较低）。

random forest属于bagging，所有predictor的训练都是平行的。而boosting都是sequential（串行）的, adaboost是boosting的其中一种。

[adaboost](url)的训练pipeline:
1. 选择一个base model（比如说decision tree classifier）来训练数据，然后使用得到的参数来预测training set。
2. 初始设置：每一个instance的权重都是1/m（如果假设一共有m个数据）。
_predictor的相对错误比例(r)_： **预测值不等于实际值的个数（误判）之和 / 现在的instance权重之和**
_更新predictor的权重_ = **learning rate* log(1-r/r)**
从上式可以看出，如果r（错误率）增加，那么权重会相应减小，反之亦反。

3. 被误判的数据个例的相对权重会被提高，以便于下一次训练时着重纠正这些部分。
4. 下一轮的训练中使用更新过的权重，然后回到第一步，重返往复。
5. 当predictors的数量达到设定的预期值，或者找到了一个完美predictor，那么整个过程就会停止。
所谓boosting是指的每一轮抬高(boost)被误判的data instance的权重（weights）。

你可能会想到梯度下降（gradient descent）也是一步步迭代。梯度下降是一直微调单一的predictor，而adaboost会在迭代时加入predictors，从而使得其更加有效。

key points: 权重的更新是predictor和instance双方的。前者错误时，权重被降低。后者错误时，权重被升高。

**Comment by emptymalei** [comment]
published in https://github.com/data-com/weekly/issues/9
